NGINX proxy for everything (#6621)
DataDog/documentation: content/en/agent/proxy.md

## Using NGINX as a Proxy

> [NGINX][3] is a web server which can also be used as a reverse proxy, load balancer, mail proxy, and HTTP cache. You can also use NGINX as a proxy for your Datadog agents:

Reviewer: ```suggestion
[NGINX][3] is a web server which can also be used as a reverse proxy, load balancer, mail proxy, and HTTP cache. You can also use NGINX as a proxy for your Datadog Agents:
```

`agent ---> nginx ---> Datadog`

### Proxy forwarding with NGINX

#### NGINX configuration

> This example `nginx.conf` can be used to proxy agent traffic to the Datadog intake. The last server block in this configuration does TLS wrapping to ensure internal plaintext logs are encrypted between your proxy and Datadog's log intake API endpoint:

Reviewer: ```suggestion
This example `nginx.conf` can be used to proxy Agent traffic to Datadog. The last server block in this configuration does TLS wrapping to ensure internal plaintext logs are encrypted between your proxy and Datadog's log intake API endpoint:
```

Do not change the `logs_no_ssl` parameter as NGINX is simply forwarding the traffic to Datadog and does not decrypt or encrypt the traffic.

> When choosing to send logs over HTTPS instruct the Agent in the following way `datadog.yaml`:

Reviewer: ```suggestion
When choosing to send logs over HTTPS, use the following code block in `datadog.yaml` to configure Agent behavior:
```

```yaml
logs_config:

>   logs_dd_url: myProxyServer.myDomain:3837

Reviewer 2: ```suggestion
  logs_dd_url: "<PROXY_SERVER_DOMAIN>:3837"
```
